### PR TITLE
Roadma, harmonization OME NGFF, field names

### PR DESCRIPTION
**Pull request by bengtl**
Roadmap added as well as harmonization with the OME NGFF metadata model: [https://ngff.openmicroscopy.org/latest/#metadata](https://ngff.openmicroscopy.org/latest/#metadata)

Also, camelCase notation was adapted for all fields.

### CODE REVIEW

File: docs/tiling.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+- **`center`:** Optional. Indicates the center and suggested default view of the tileset. Minimum of three numbers in the order `[x, y, zoom]`, but may include up to a further three numbers for a total of six, `[t,c,z,x,y,zoom]`, where `t` is the tileset timestamp, `c` is the channel, `z` is the z coordinate, `x` and `y` are the x and y coordinates, and `zoom` is the zoom level. Zoom level should be last.
+- **`vector_layers`:**  Required. Describes each layer within the vector tiles, and has the following structure:
+
+  - **`id`:** Required. A unique identifier for the layer. Required for each layer.
+  - **`fields`:** Required. A list of fields (attributes) and their data types. For MicroJSON, this can either be an empty list, or any combination of `string`, `numerical`, and `multiNumeric`, attributes with their sub-attributes, as described in the MicroJSON specification. Required for each layer.
+  - **`description`:** Optional. A brief description of the layer.
+  - **`minzoom` and `maxzoom`:** Optional. The range of zoom levels at which the layer is visible.
```
Comment (hamshkhawar) [inline docs/tiling.md:26]: Since you are using camelcase format,
minzoom should be minZoom in that case

Comment (hamshkhawar) [reply docs/tiling.md:26]: Same is true for all others

File: docs/tiling.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                    "image": "Image URI"
+                },
+                "numerical": {
+                    "label": "Label ID",
+                    "channel": "Channel ID"
+                }
+            }
+        }
+    ],
+    "fillzoom": 3
```
Comment (hamshkhawar) [inline docs/tiling.md:83]: fillZoom

File: docs/tiling.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    "attribution": "<a href='http://example.com'>Example</a>",
+    "tiles": [
+        "http://example.com/tiled_example/{zlvl}/{x}/{y}.json"
+    ],
+    "minzoom": 0,
+    "maxzoom": 10,
+    "bounds": [0, 0, 24000, 24000],
+    "center": [12000, 12000, 0],
+    "format": "json",
+    "vector_layers": [
```
Comment (hamshkhawar) [inline docs/tiling.md:105]: "vector_layers" -->  vectorLayers

File: docs/tiling.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        ]
+      },
+      "properties": {
+        "numeric": {
+          "label": 3
+        }
+      }
+    }
+  ],
+  "multiscale": {
```
Comment (hamshkhawar) [inline docs/tiling.md:169]: "multiscale" multiScale

File: src/microjson/tile.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+from typing import List, Optional, Union, Dict
+from pydantic import BaseModel, AnyUrl, conlist, RootModel
+from pathlib import Path
+
+
+class TileLayer(BaseModel):
+    id: str
+    fields: Union[None, Dict[str, str], Dict[str, Dict]]
+    minzoom: Optional[int] = None
```
Comment (hamshkhawar) [inline src/microjson/tile.py:9]: change minzoom and maxzoom to Camel case

File: src/microjson/utils.py [hunk 6]
```diff
@@ -390,69 +396,74 @@ def polygons_to_microjson(  # noqa: PLR0915
...
-        y= int(meta.string.get("Y"))
+        y = int(meta.string.get("Y"))
         fmask = np.zeros((x, y), dtype=np.uint8)
         for i, _ in enumerate(poly):
             image = fmask.copy()
             pol = np.array(poly[i][0])
             mask = sk.draw.polygon2mask((x, y), pol)
-            image[mask == False] = 0
-            image[mask == True] = 1
+            image[not mask] = 0
```
Comment (hamshkhawar) [inline src/microjson/utils.py:458]: Have you tested this change? I wonder may be this will break?

File: tests/json/tilejson/valid/minimal.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+    "tilejson": "3.0.0",
+    "tiles": [
+        "http://localhost:8080/tiles/{zlvl}/{x}/{y}.json"
+    ],
+    "vector_layers": [
```
Comment (hamshkhawar) [inline tests/json/tilejson/valid/minimal.json:6]: "vector_layers" change to camel case

File: tests/json/tilejson/valid/osm.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    "tiles": [
+        "https://a.tile.custom-osm-tiles.org/{z}/{x}/{y}.mvt",
+        "https://b.tile.custom-osm-tiles.org/{z}/{x}/{y}.mvt",
+        "https://c.tile.custom-osm-tiles.org/{z}/{x}/{y}.mvt"
+    ],
+    "minzoom": 0,
+    "maxzoom": 18,
+    "bounds": [ -180, -85, 180, 85 ],
+    "fillzoom": 6,
+    "something_custom": "this is my unique field",
```
Comment (hamshkhawar) [inline tests/json/tilejson/valid/osm.json:17]: change all snake case to camel case format

**Review by hamshkhawar — CHANGES_REQUESTED** [review]
@bengtl I have looked at this PR and left some comments. Mostly using camel case format for some of the attributes.

File: docs/example.md [hunk 9]
```diff
@@ -55,25 +53,23 @@ This JSON file demonstrates how MicroJSON can be used to define and describe dif
             "max": 1
         }
     },
-    "string_fields": ["well","imagename"],
-    "coordinatesystem": {
+    "descriptiveFields": ["well","imagename"],
+    "multiscale": {
```
[the remark this answers is not in the document]
Comment (bengtl) [reply docs/example.md:57]: multiscale without capitalization as in camelCase comes from OME NGFF metadata model.

File: docs/example.md [hunk 10]
```diff
@@ -55,25 +53,23 @@ This JSON file demonstrates how MicroJSON can be used to define and describe dif
...
       "axes": [
           {
               "name": "x",
               "unit": "micrometer",
-              "type": "cartesian",
-              "pixelsPerUnit": 1,
+              "type": "space",
```
[the remark this answers is not in the document]
Comment (bengtl) [reply docs/example.md:62]: No, it is optional. Type also comes from OME NGFF.

File: docs/metadata_example.md [hunk 11]
```diff
@@ -1,14 +1,14 @@
 # Metadata Usage in MicroJSON Example
 
-This guide demonstrates how to populate the `string`, `numeric`, and `multi_numeric` fields under the 'properties' field as per the MicroJSON specification using a rectangular polygon as the geometry.
+This guide demonstrates how to populate the `string`, `numeric`, and `multiNumeric` fields under the 'properties' field as per the MicroJSON specification using a rectangular polygon as the geometry.
```
[the remark this answers is not in the document]
Comment (bengtl) [reply docs/metadata_example.md:3]: I think under is correct.

File: docs/metadata_example.md [hunk 12]
```diff
@@ -1,14 +1,14 @@
...
 
 ## Properties Class Overview
 
 In MicroJSON, metadata related to a feature is stored in the `Properties` class. This class encompasses three fields:
 
 - `string`: A dictionary containing string metadata. This can include textual information such as the name and description of the feature.
 - `numeric`: A dictionary containing numeric metadata. This field can be used to store quantifiable data related to the feature, like the cell count in a microscopic image.
-- `multi_numeric`: A dictionary where each entry is a list of numeric values. This can be useful for storing data with multiple numeric values under a single key, such as ratio of infectivity over time.
+- `multiNumeric`: A dictionary where each entry is a list of numeric values. This can be useful for storing data with multiple numeric values under a single key, such as ratio of infectivity over time.
```
[the remark this answers is not in the document]
Comment (bengtl) [reply docs/metadata_example.md:11]: No, under is correct, see for example: https://www.e-education.psu.edu/ngapython/node/824

File: docs/index.md [hunk 13]
```diff
@@ -8,9 +8,9 @@ MicroJSON is a format, inspired by [GeoJSON](https://geojson.org), for encoding
...
 
-- `"coordinatesystem"`: (Optional) A coordinate system object as defined in the section [Coordinates object](#coordinates-object). If this property is not present, the default coordinate system is assumed to be the same as the image coordinate system, using cartesian coordinates and pixels as units. It is recommended to define this property at the top level of the MicroJSON object, but it MAY also be defined at the level of a Feature or Geometry object, in which case it overrides the top level coordinate system.
+- `"multiscale"`: (Optional) A multiscale object as defined in the section [Multiscale object](#multiscale-object). If this property is not present, the default coordinate system is assumed to be the same as the image coordinate system, using cartesian coordinates and pixels as units. It is recommended to define this property at the top level of the MicroJSON object, but it MAY also be defined at the level of a Feature or Geometry object, in which case it overrides the top level coordinate system.
```
[the remark this answers is not in the document]
Comment (bengtl) [reply docs/index.md:13]: changed

File: docs/tiling.md (hunk 4 above)
Comment (bengtl) [reply docs/tiling.md:169]: multiscale without capitalization as in camelCase comes from OME NGFF metadata model.

File: src/microjson/tile.py (hunk 5 above)
Comment (bengtl) [reply src/microjson/tile.py:9]: They are harmonized with TileJSON terminology.

File: src/microjson/utils.py (hunk 6 above)
Comment (bengtl) [reply src/microjson/utils.py:458]: I had some other problems with the test case. I will revert this change for now, I think you are right.

File: tests/json/tilejson/valid/minimal.json (hunk 7 above)
Comment (bengtl) [reply tests/json/tilejson/valid/minimal.json:6]: vector_layers without capitalization as in camelCase comes from TileJSON.

**Pull request closed without merging**